src/backend/utils/storage.js

```diff
@@ -1,16 +1,38 @@
+const crypto = require('crypto');
+const normalizeUrl = require('normalize-url');
+const { logger } = require('./logger');
 const { redis } = require('../lib/redis');
 
+const standardize = (url, type) => {
+  let namespace = 't:post:';
+  let processed = url;
+
+  if (type === 'feed') {
```

Comment:
This feels brittle to me. Can we separate out the `normalizeUrl(url)` call, so that it happens before we pass in the URL to this function?  That way you don't need to worry about it here, and the caller can decide if it's needed.

Reply:
One way to keep this a single function call would be to do:

```js
function createPostKey(uri) {
  const normalizedUri = normalizeUrl(url);
  return createKey(normalizedUri, 'post');
}

function createFeedKey(uri) {
  return createKey(uri, 'feed');
}
```

And let the common bits happen in `createKey()`.

Reply:
Added your suggestion, definitely clearer and easier to understand. Thank you!